unified-runtime/source/adapters/level_zero/v2/queue_immediate_in_order.cpp

```diff
@@ -712,25 +716,106 @@ ur_result_t ur_queue_immediate_in_order_t::enqueueWriteHostPipe(
   return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
 }
 
+ur_result_t ur_queue_immediate_in_order_t::enqueueUSMAllocHelper(
+    ur_queue_handle_t hQueue, ur_usm_pool_handle_t pPool, const size_t size,
+    const ur_exp_async_usm_alloc_properties_t *, uint32_t numEventsInWaitList,
+    const ur_event_handle_t *phEventWaitList, void **ppMem,
+    ur_event_handle_t *phEvent, ur_usm_type_t type) {
+  auto commandListLocked = commandListManager.lock();
+
+  if (!pPool) {
+    pPool = hContext->getAsyncPool();
+  }
+
+  auto device = (type == UR_USM_TYPE_HOST) ? nullptr : hDevice;
+
+  std::vector<ur_event_handle_t> extendedWaitList;
```

Comment:
You don;t need to create a temporary vector here. commandListManger has getWaitListView function that you should use here (it also accepts an optional extra wait event). You just call: `getWaitListView(commandListLocked, phEventWaitList,  numEventsInWaitList, originAllocEvent);`